velox/exec/tests/utils/QueryAssertions.cpp

```diff
@@ -253,8 +257,9 @@ velox::variant variantAt<TypeKind::HUGEINT>(
     ::duckdb::DataChunk* dataChunk,
     int32_t row,
     int32_t column) {
-  auto hugeInt = ::duckdb::HugeIntValue::Get(dataChunk->GetValue(column, row));
-  return velox::variant(HugeInt::build(hugeInt.upper, hugeInt.lower));
+  auto unscaledValue =
```

Comment:
What is the issue with `auto hugeInt = ::duckdb::HugeIntValue::Get(dataChunk->GetValue(column, row));`?

Reply:
This `auto hugeInt = ::duckdb::HugeIntValue::Get(dataChunk->GetValue(column, row));` approach leads to downcast of integer inside DuckDb at somepoint and leads to wrong results. I remember debugging this while working `DuckDb vector` to `Velox vector` conversion.

Here is what we see at the time of comparison and leads to mismatch.
```
(lldb) p *(__int128_t *)(0x60000349ecb0) -> Actual
(__int128_t) -36893488147419103235
(lldb) p (__int128_t *)(0x6000034b9c80)
(__int128_t *) 0x00006000034b9c80
(lldb) p *(__int128_t *)(0x6000034b9c80) -> Expected in duckDB.
(__int128_t) 112571972338393865790930626012
```

Reply:
Hi @karteekmurthys, could you elaborate a bit where the downcast happens? I found that ::duckdb::HugeIntValue::Get() returns value_.hugeint directly whereas GetValue<::duckdb::hugeint_t>() does `Cast::Operation<hugeint_t, hugtint_t>(value_.hugeint)`. So feels like they should be the same if values in dataChunk are hugeint_t.

Reply:
@karteekmurthys, In case the comment above was hidden, just FYI, I also left a question here yesterday.

Reply:
I had encountered this downcast somewhere in duckdb code before as well. Earlier, we were copying duckdb code as part of velox and we could drill down into duckdb in gdb. But, it is not possible now since we include the library. However, like I have shared here, the value is corrupted without my fix.
```
(lldb) p *(__int128_t *)(0x60000349ecb0) -> Actual (__int128_t) -36893488147419103235 (lldb) p (__int128_t *)(0x6000034b9c80) (__int128_t *) 0x00006000034b9c80 (lldb) p *(__int128_t *)(0x6000034b9c80) -> Expected in duckDB. (__int128_t) 112571972338393865790930626012
```